app/controllers/application_controller.rb

```diff
@@ -237,10 +238,9 @@ def respond_to_access_denied
   def redirection_url
     # Try to redirect somewhere sensible. Note: not all controllers have an index action
     url = if current_user.present?
-      (respond_to?(:index) and self.action_name != 'index') ? { action: 'index' } : root_url
-    else
-      login_url
+            (respond_to?(:index) && action_name != 'index') ? { action: 'index' } : root_url
```

Comment:
Line is too long. [92/80]<br>Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.